### PR TITLE
refactor: split `SimpleGraph.Connectivity` into 5 files

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Refactoring.20the.20graph.20theory.20library

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by adomani — APPROVED** [review]
LGTM!

I am not commenting about the appropriateness of the splitting, but I am very much in favour of smaller files!

I checked that all lemmas/theorems/defs were preserved in the process.
I have not checked whether some signatures may have changed: this is possible, since there were a few different `variable`s commands, but it is probably not a big deal.

The only reason that I am not `maintainer merge`ing this PR is that I am not too familiar with the Graph Theory part of the library.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by kmill** [comment]
Sorry to do this:

bors r-

I really want a chance to review how this was split. I've been busy the last 24 hours with non-mathlib work, and I have questions and comments about the split that I'll need time to formulate.

**Comment by jcommelin** [comment]
> I really want a chance to review how this was split. I've been busy the last 24 hours with non-mathlib work, and I have questions and comments about the split that I'll need time to formulate.

@kmill Could you please give an update with those questions and comments?

File: Mathlib/Combinatorics/SimpleGraph/Connectivity/Walk.lean [hunk 1]
```diff
@@ -0,0 +1,1376 @@
...
+/-- Predicate for the empty walk.
+
+Solves the dependent type problem where `p = G.Walk.nil` typechecks
+only if `p` has defeq endpoints. -/
+inductive Nil : {v w : V} → G.Walk v w → Prop
+  | nil {u : V} : Nil (nil : G.Walk u u)
+
+@[simp] lemma nil_nil : (nil : G.Walk u u).Nil := Nil.nil
+
+@[simp] lemma not_nil_cons {h : G.Adj u v} {p : G.Walk v w} : ¬(cons h p).Nil := fun.
```
Comment (kmill) [inline Mathlib/Combinatorics/SimpleGraph/Connectivity/Walk.lean:782]: This fixes it:
```suggestion
@[simp] lemma not_nil_cons {h : G.Adj u v} {p : G.Walk v w} : ¬(cons h p).Nil := nofun
```

**Pull request closed without merging**